include/sacn/cpp/source.h

```diff
@@ -678,65 +710,107 @@ inline int Source::ProcessManual()
+ * @param[in, out] netints If !empty, this is the list of interfaces the application wants to use, and the status codes
+ * are filled in.  If empty, all available interfaces are tried and this vector isn't modified.
  * @return #kEtcPalErrOk: Networking reset successfully.
  * @return #kEtcPalErrNoNetints: None of the network interfaces provided were usable by the library.
- * @return #kEtcPalErrInvalid: Invalid parameter provided.
  * @return #kEtcPalErrNotInit: Module not initialized.
- * @return #kEtcPalErrNotFound: Handle does not correspond to a valid source, or the universe was not found on this
- *                              source.
  * @return #kEtcPalErrSys: An internal library or system call error occurred.
  */
-inline etcpal::Error Source::ResetNetworking(uint16_t universe, std::vector<SacnMcastInterface>& netints)
+inline etcpal::Error Source::ResetNetworking(std::vector<SacnMcastInterface>& netints)
 {
   if (netints.empty())
-    return sacn_source_reset_networking(handle_, universe, nullptr, 0);
+    return sacn_source_reset_networking(nullptr, 0);
+
+  return sacn_source_reset_networking(netints.data(), netints.size());
+}
 
-  return sacn_source_reset_networking(handle_, universe, netints.data(), netints.size());
+/**
+ * @brief Resets the underlying network sockets and determines network interfaces for each universe of each source.
+ *
+ * This is typically used when the application detects that the list of networking interfaces has changed, and wants to
+ * determine what the new network interfaces should be for each universe of each source.
+ *
+ * After this call completes successfully, all universes of all sources are considered to be updated and have new values
+ * and priorities. It's as if every source just started sending values on all their universes.
+ *
+ * If this call fails, the caller must call Shutdown() on all sources, because the source API may be in an
+ * invalid state.
+ *
+ * Note that the networking reset is considered successful if it is able to successfully use any of the
+ * network interfaces passed in for each universe.  This will only return #kEtcPalErrNoNetints if none of the interfaces
+ * work for a universe.
+ *
+ * @param[in, out] netint_lists Vector of lists of interfaces the application wants to use for each universe. Must not
+ * be empty. Must include all universes of all sources, and nothing more. The status codes are filled in whenever
+ * UniverseNetintList::netints is !empty.
+ * @return #kEtcPalErrOk: Networking reset successfully.
+ * @return #kEtcPalErrNoNetints: None of the network interfaces provided for a universe were usable by the library.
+ * @return #kEtcPalErrInvalid: Invalid parameter provided.
+ * @return #kEtcPalErrNotInit: Module not initialized.
+ * @return #kEtcPalErrSys: An internal library or system call error occurred.
+ */
+inline etcpal::Error Source::ResetNetworking(std::vector<UniverseNetintList>& netint_lists)
```

Comment:
I see why you added this function for completeness, but I don't know that anyone's use case will be this complex. We might want to leave it out until someone asks for it. It took me a bit to wrap my head around how to use it...

Reply:
This is the only resetnetworking version that allows you to set different network interfaces for the different sources.  Since that was a request for source creation, we may need to keep this version, right?

Reply:
Yeah, we do. Never mind.

Reply:
We resolved this offline.